src/mod_websockets.erl

```diff
@@ -358,3 +358,38 @@ maybe_send_ping_request(none) ->
     ok;
 maybe_send_ping_request(PingRate) ->
     send_ping_request(PingRate).
+
+
+%%--------------------------------------------------------------------
+%% add_sec_websocket_protocol_header
+%%--------------------------------------------------------------------
+
+add_sec_websocket_protocol_header(Req) ->
+     case cowboy_req:header(<<"sec-websocket-protocol">>, Req) of
+         undefined ->
+             %% Server should not set "sec-websocket-protocol" respond header
+             %% if it does not present in the request
+             Req;
+         Proto ->
+             %% A client can propose several protocols
+             %% but we are interested in xmpp only
+             Protocols = split_protocol(Proto),
+             case lists:member(<<"xmpp">>, Protocols) of
+                 true ->
+                     cowboy_req:set_resp_header(<<"Sec-WebSocket-Protocol">>, <<"xmpp">>, Req);
+                 false ->
+                     %% Do not agree with client, do not add a respond header
+                     ?DEBUG("issue=not_supported_procol, protocol=~p", [Proto]),
```

Comment:
Is it enough to not set the resp header maybe an error should be send to the client? Is it documented somewhere?

Reply:
Documented. We reply with nil or no header. Client can continue on it’s own risk or disconnect